skrub/_drop_uninformative.py

```diff
@@ -0,0 +1,172 @@
+        count as one additional distinct value.
+
+    drop_if_id : bool, default=False
+        If True, drop the column if all values are distinct. Missing values count as
+        one additional distinct value. Numeric columns are never dropped.
+
+    drop_null_fraction : float or None, default=1.0
+        Drop columns with a fraction of missing values larger than threshold. If None,
+        keep the column even if all its values are missing.
+
+    Examples
+    --------
+    >>> from skrub import DropUninformative
+    >>> import pandas as pd
+    >>> df = pd.DataFrame({"col1": [None, None, None]})
+
+    By default, only null columns are dropped:
+
+    >>> du = DropUninformative()
+    >>> du.fit_transform(df["col1"])
+    []
+
+    It is also possible to drop constant columns, or specify a lower null fraction
+    threshold:
+
+    >>> df = pd.DataFrame({"col1": [1,2,None], "col2": ["const", "const", "const"]})
+    >>> du = DropUninformative(drop_if_constant=True, drop_null_fraction=0.1)
+    >>> du.fit_transform(df["col1"])
+    []
+    >>> du.fit_transform(df["col2"])
+    []
+
+    Finally, it is possible to set ``drop_if_id`` to ``True`` in order to drop
+    string columns that contain all distinct values:
+
+    >>> df = pd.DataFrame({"col1": ["A", "B", "C"]})
+    >>> du = DropUninformative(drop_if_id=True)
+    >>> du.fit_transform(df["col1"])
+    []
+    """
+
+    def __init__(
+        self,
+        drop_if_constant=False,
+        drop_if_id=False,
+        drop_null_fraction=1.0,
+    ):
+        self.drop_if_constant = drop_if_constant
+        self.drop_if_id = drop_if_id
+        self.drop_null_fraction = drop_null_fraction
+
+    def _check_params(self):
+        if not isinstance(self.drop_if_constant, bool):
+            raise TypeError(
+                f"drop_if_constant must be boolean, found {self.drop_if_constant}."
+            )
+        if not isinstance(self.drop_if_id, bool):
+            raise TypeError(f"drop_if_id must be boolean, found {self.drop_if_id}.")
+
+        if self.drop_null_fraction is not None:
+            if (
+                not isinstance(self.drop_null_fraction, numbers.Number)
+                or not 0.0 <= self.drop_null_fraction <= 1.0
+            ):
+                raise ValueError(
+                    f"Threshold {self.drop_null_fraction} is invalid. Threshold"
+                    " should be a number in the range [0, 1], or None."
+                )
+
+    def _drop_if_too_many_nulls(self, column):
+        if self.drop_null_fraction == 1.0:
+            return self._null_count == len(column)
+        # No nulls found, or no threshold
+        if self._null_count == 0 or self.drop_null_fraction is None:
+            return False
+        return self._null_count / len(column) > self.drop_null_fraction
+
+    def _drop_if_constant(self, column):
+        if self.drop_if_constant:
+            if (sbd.n_unique(column) == 1) and (not self._null_count > 0):
```

Comment:
Nitpick: simplify
```suggestion
            if (sbd.n_unique(column) == 1) and (self._null_count == 0):
```

Reply:
If we want, this can be simplified further as:
```python
return self.drop_if_constant and (sbd.n_unique(column) == 1) and (self._null_count == 0)
```
without hurting readability